test/puppetlabs/puppetdb/integration/smoke.clj

```diff
@@ -27,3 +27,30 @@
           (is (= catalog-uuid report-uuid)))))))
 
 
+(deftest ^:integration rich-data-agent-run
+  (with-open [pg (int/setup-postgres)
+              pdb (int/run-puppetdb pg {})
+              ps (int/run-puppet-server [pdb] {:agent {:rich-data true}})]
+
+    (testing "Agent run succeeds"
+      (let [{:keys [out]} (int/run-puppet-as "my-agent" ps pdb "notify { 'foo':\n  message => {\n  'regex'     => /[a-z]+/,\n  'object'    => Object[{name => 'FindMe', 'attributes' => {'x' => String}}],\n  'version'   => SemVer('3.4.1'),\n  'semver'    => SemVerRange('>=3.4.1'),\n  'timestamp' => Timestamp('1994-03-25T19:30:00'),\n  'timespan'  => Timespan('3-10:00:00'),\n  }\n}" {:rich-data true})]
```

Comment:
maybe break the manifest string across multiple lines so it's easier to read. Something like: 
 (let [manifest "notify { 'foo':\n
                                 message =>
                                 {\n  'regex'     => /[a-z]+/,
                                  \n  'object'    => Object[{name => 'FindMe', 'attributes' => {'x' => String}}],
                                  \n  'version'   => SemVer('3.4.1'),
                                   ... "])

When testing this locally it also looks like it works without the \n newline characters so maybe we can drop those out too.